### PR TITLE
[Enhancement] PseudoBackend support report task; add failure injection and write quorum test

### PR DESCRIPTION
**Pull request by decster**
## What type of PR is this：
- [ ] bug
- [ ] feature
- [x] enhancement
- [ ] refactor
- [ ] others

## Problem Summary(Required) ：

The PR adds task report functionality to PseudoBackend, and adds write failure injection to test txn's quorum write logic.

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by decster** [comment]
run starrocks_admit_test

File: fe/fe-core/src/test/java/com/starrocks/transaction/TxnWithWriteFailureTest.java [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+            System.out.printf("numLoad:%d Time: %.2fs, %.2f tps\n", numLoad, t, numLoad / t);
+        } finally {
+            cluster.getBackend(10001).setWriteFailureRate(0.0f);
+        }
+    }
+
+    @Test(expected = SQLException.class)
+    public void test3Replica2FailInsertFail() throws Exception {
+        PseudoCluster cluster = PseudoCluster.getInstance();
+        cluster.getBackend(10001).setWriteFailureRate(1.0f);
```
Comment (chaoyli) [inline fe/fe-core/src/test/java/com/starrocks/transaction/TxnWithWriteFailureTest.java:62]: setWriteFailureRate(2.0f) ?

Comment (decster) [reply fe/fe-core/src/test/java/com/starrocks/transaction/TxnWithWriteFailureTest.java:62]: rate is 1.0 at max, why set to 2.0?

**Review by chaoyli — APPROVED** [review]

**Review by meegoo — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Pull request merged**